fitlins/cli/run.py

```diff
@@ -170,18 +168,53 @@ def run_fitlins(argv=None):
         pipeline_name += '_' + opts.derivative_label
     deriv_dir = op.join(opts.output_dir, pipeline_name)
     os.makedirs(deriv_dir, exist_ok=True)
-
     bids.write_derivative_description(opts.bids_dir, deriv_dir)
 
     work_dir = mkdtemp() if opts.work_dir is None else opts.work_dir
 
+    # Go ahead and initialize the layout database
+    if opts.database_path is None:
+        database_path = Path(work_dir) / 'dbcache'
+        reset_database = True
+        make_layout = True
+    elif Path(opts.database_path).exists():
+        layout = BIDSLayout.load(opts.database_path)
+        database_path = opts.database_path
+        make_layout = False
+    else:
+        database_path = opts.database_path
+        reset_database = False
+        make_layout = True
+
+    if make_layout:
+        layout = BIDSLayout(opts.bids_dir,
+                            derivatives=derivatives,
+                            ignore=opts.ignore,
+                            validate=False,
+                            force_index=opts.force_index,
+                            database_path=database_path,
+                            reset_database=reset_database)
+
+    subject_list = None
+    if opts.participant_label is not None:
+        subject_list = bids.collect_participants(
+            opts.bids_dir, participant_label=opts.participant_label,
+            database_path=database_path)
```

Comment:
Should we just pass the layout, rather than instantiating another one in the function?

Reply:
Looks like it